src/main/java/heronarts/lx/output/LXBufferOutput.java

```diff
@@ -127,7 +128,7 @@ protected LXBufferOutput updateDataBuffer(int[] colors, byte[][] glut, double br
             int r = ((color >> 16) & 0xff);
             int g = ((color >> 8) & 0xff);
             int b = (color & 0xff);
-            int w = (r < g) ? ((r < b) ? r : b) : ((g < b) ? g : b);
+            int w = LXUtils.constrain((r + b + g) / 3, 0, 255);
```

Comment:
Drop the LXUtils.constrain call here... there is no way that this value can be under 0 or over 255, it won't matter on most projects but that's potentially a ton of extra function calls doing nothing for huge pixel count projects.

The only thing I might consider instead is a Math.round(), but I am inclined to just leave it off. If people are sending W-only then they should generally be animating patterns in gray, in which case r=g=b anyways and there will be no rounding errors.

Nuke the function call and we're good to go here - thank you for catching this, definitely an oversight on my part!

Reply:
Good lesson there.  Burning CPU cycles when I should have checked to see if it was necessary.